### PR TITLE
[Core] RGB Matrix limit basic indicators to the last render

### PR DESCRIPTION
**Pull request by Xelus22**
## Description

Only run the basic indicators on the last render so you don't run it unnecessarily.

By default we have 5 iterations of the render state and this limits it to a single iteration of the render cycle.

Gives negligble scan rate gain still an optimization :P
tested on teleport/native/iso by @ebastler
image

Future problem:
When an indicator is on/updated, the driver update flag is set dirty. Then on the next cycle, the animation will render and see the colour is different and set the dirty flag. Which then if the indicator LED is still on, it will set to the correct colour and set the dirty flag again. Thus even between cycles when an indicator is on, it will always be set dirty and flushed.
This is the reason for such a large performance hit when an indicator is on compared to when it is just the animation

- [x] Core
- [ ] Bugfix
- [ ] New feature
- [x] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* Decrease wasted render cycles of the indicators.

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by zvecr** [comment]
Same change should be made to led_matrix, as its functionally the same in this area.

File: quantum/led_matrix/led_matrix.h [hunk 1]
```diff
@@ -43,6 +43,7 @@
 
 #ifndef LED_MATRIX_LED_PROCESS_LIMIT
 #    define LED_MATRIX_LED_PROCESS_LIMIT (LED_MATRIX_LED_COUNT + 4) / 5
+#    define LED_MATRIX_LED_PROCESS_MAX_ITERATIONS ((LED_MATRIX_LED_COUNT + LED_MATRIX_LED_PROCESS_LIMIT - 1) / LED_MATRIX_LED_PROCESS_LIMIT)
 #endif
```
Comment (zvecr) [inline quantum/led_matrix/led_matrix.h:47]: Im guessing we do not expect users to have to also specify `LED_MATRIX_LED_PROCESS_MAX_ITERATIONS` when configuring `LED_MATRIX_LED_PROCESS_LIMIT`? As right now the code will fail to compile when `LED_MATRIX_LED_PROCESS_LIMIT` is not defined.
```suggestion
#endif
#define LED_MATRIX_LED_PROCESS_MAX_ITERATIONS ((LED_MATRIX_LED_COUNT + LED_MATRIX_LED_PROCESS_LIMIT - 1) / LED_MATRIX_LED_PROCESS_LIMIT)
```

File: quantum/rgb_matrix/rgb_matrix.h [hunk 2]
```diff
@@ -49,7 +49,8 @@
 #endif
 
 #ifndef RGB_MATRIX_LED_PROCESS_LIMIT
-#    define RGB_MATRIX_LED_PROCESS_LIMIT (RGB_MATRIX_LED_COUNT + 4) / 5
+#    define RGB_MATRIX_LED_PROCESS_LIMIT ((RGB_MATRIX_LED_COUNT + 4) / 5)
+#    define RGB_MATRIX_LED_PROCESS_MAX_ITERATIONS ((RGB_MATRIX_LED_COUNT + RGB_MATRIX_LED_PROCESS_LIMIT - 1) / RGB_MATRIX_LED_PROCESS_LIMIT)
 #endif
```
Comment (zvecr) [inline quantum/rgb_matrix/rgb_matrix.h:54]:
```suggestion
#endif
#define RGB_MATRIX_LED_PROCESS_MAX_ITERATIONS ((RGB_MATRIX_LED_COUNT + RGB_MATRIX_LED_PROCESS_LIMIT - 1) / RGB_MATRIX_LED_PROCESS_LIMIT)
```

File: quantum/led_matrix/led_matrix.h (hunk 1 above)
Comment (Xelus22) [reply quantum/led_matrix/led_matrix.h:47]: Ah yes it should be internal yep.

File: quantum/led_matrix/led_matrix.h [hunk 3]
```diff
@@ -44,6 +44,7 @@
 #ifndef LED_MATRIX_LED_PROCESS_LIMIT
 #    define LED_MATRIX_LED_PROCESS_LIMIT (LED_MATRIX_LED_COUNT + 4) / 5
```
Comment (zvecr) [inline quantum/led_matrix/led_matrix.h:45]: Just to align changes
```suggestion
#    define LED_MATRIX_LED_PROCESS_LIMIT ((LED_MATRIX_LED_COUNT + 4) / 5)
```

**Review by zvecr — APPROVED** [review]

**Review by drashna — APPROVED** [review]

**Comment by filterpaper** [comment]
The following configuration uses the LEDs on a pair of KB2040 for RGB matrix:
```c
#define WS2812_DI_PIN 17U
#define WS2812_PIO_USE_PIO1
#define RGB_MATRIX_LED_COUNT 2
#define RGB_MATRIX_SPLIT {1, 1}
```
`RGB_MATRIX_LED_PROCESS_LIMIT` will be calculated as `1` with this PR, and LED on the _left side_ of the split will not light up for `rgb_matrix_indicators_user()` function. The left side `iter` count stops at `1` and does not reach `2`.

Behaviour can be resolved by explicitly defining `#define RGB_MATRIX_LED_PROCESS_LIMIT 2`. This PR is not the root cause. It might have exposed a corner use case of two LEDs setup in this manner.

**Comment by kilinrax** [comment]
> Behaviour can be resolved by explicitly defining `#define RGB_MATRIX_LED_PROCESS_LIMIT 2`.

This does not resolve the problem (all left side LEDs stay red) on a Kyria rev 3.  Reverting commit c4f66e5d6e57d2100a8d3a08758075c470f04767 does.

**Pull request merged**